libsolidity/analysis/PostTypeContractLevelChecker.cpp

```diff
@@ -101,29 +102,61 @@ void PostTypeContractLevelChecker::checkStorageLayoutSpecifier(ContractDefinitio
 	}
 
 	auto const* baseSlotExpressionType = type(baseSlotExpression);
-	auto const* rationalType = dynamic_cast<RationalNumberType const*>(baseSlotExpressionType);
-	if (!rationalType)
+	if (
+		!dynamic_cast<IntegerType const*>(baseSlotExpressionType) &&
+		!dynamic_cast<RationalNumberType const*>(baseSlotExpressionType)
+	)
 	{
 		m_errorReporter.typeError(
 			6396_error,
 			baseSlotExpression.location(),
-			"The base slot of the storage layout must evaluate to a rational number."
+			"The base slot of the storage layout must evaluate to a rational integer number."
```

Comment:
```suggestion
			"The base slot of the storage layout must evaluate to an integer."
```

We could also say `a rational number or an integer`, but not sure if the user will get the distinction. Unfortunately, out terminology here is a bit ambiguous.